src/api/app/jobs/send_event_emails.rb

```diff
@@ -1,22 +1,29 @@
+# TODO: Rename this to something more appropriate like "ProcessEvents"
 class SendEventEmails
   # we don't need this outside of the migration - but we need to be able
   # to load old jobs from the database (and mark their events) before deleting
   # (see 20151030130011_mark_events)
   attr_accessor :event
 
   def perform
-    Event::Base.where(mails_sent: false).order(:created_at).limit(1000).lock(true).each do |event|
+    Event::Base.where(mails_sent: false).order(:created_at).limit(1000).each do |event|
       event.mails_sent = true
       begin
         event.save!
       rescue ActiveRecord::StaleObjectError
         # if someone else saved it too, better don't continue - we're not alone
-        return false
+        next
       end
       subscribers = event.subscribers
       next if subscribers.empty?
+
       EventMailer.event(subscribers, event).deliver_now
+
       create_rss_notifications(event)
+
+      if needs_logging?(event)
+        create_project_log_entry(event)
```

Comment:
One liner?

```ruby
create_project_log_entry(event) if needs_logging?(event)
```

And if you implement what I commented also it could be
```ruby
event.create_log_entry
```